bitnami/etcd/templates/cronjob-defrag.yaml

```diff
@@ -0,0 +1,100 @@
+          {{- end }}
+          {{- if .Values.defrag.cronjob.nodeSelector }}
+          nodeSelector: {{- include "common.tplvalues.render" (dict "value" .Values.defrag.cronjob.nodeSelector "context" $) | nindent 12 }}
+          {{- end }}
+          {{- if .Values.defrag.cronjob.tolerations }}
+          tolerations: {{- include "common.tplvalues.render" (dict "value" .Values.defrag.cronjob.tolerations "context" $) | nindent 12 }}
+          {{- end }}
+          {{- if .Values.defrag.cronjob.serviceAccountName }}
+          serviceAccountName: {{ .Values.defrag.cronjob.serviceAccountName | quote }}
+          {{- end }}
+          containers:
+            - name: etcd-defrag
+              image: {{ include "etcd.image" . }}
+              imagePullPolicy: "IfNotPresent"
+              {{- if .Values.containerSecurityContext.enabled }}
+              securityContext: {{- include "common.compatibility.renderSecurityContext" (dict "secContext" .Values.containerSecurityContext "context" $) | nindent 16 }}
+              {{- end }}
+              env:
+                - name: BITNAMI_DEBUG
+                  value: {{ ternary "true" "false" (or .Values.image.debug .Values.diagnosticMode.enabled) | quote }}
+              {{- if .Values.defrag.cronjob.command }}
+              command: {{ .Values.defrag.cronjob.command | toYaml | nindent 16 }}
+              {{- else }}
+              command: ["etcdctl"]
+              {{- end }}
+              {{- if .Values.defrag.cronjob.args }}
+              args: {{ .Values.defrag.cronjob.args | toYaml | nindent 16 }}
+              {{- else }}
+              args:
+                - defrag
+                - --cluster
+                {{- $etcdFullname := include "common.names.fullname" . }}
+                {{- $etcdHeadlessServiceName := (printf "%s-%s" $etcdFullname "headless" | trunc 63 | trimSuffix "-") }}
+                {{- $namespace := .Release.Namespace }}
+                {{- $clusterDomain := .Values.clusterDomain }}
+                {{- $port := int .Values.service.ports.client }}
+                {{- $endpointStr := "--endpoints=[" }}
+                {{- $replicaCount := (int .Values.replicaCount) }}
+                {{- $releaseNamespace := .Release.Namespace }}
+                {{- range $iEndpoint, $e := until $replicaCount }}
+                {{- $endpoint := printf "http://%s-%d.%s.%s.svc.%s:%d" $etcdFullname $iEndpoint $etcdHeadlessServiceName $namespace $clusterDomain $port }}
+                {{- $endpointStr = printf "%s%s" $endpointStr $endpoint }}
+                {{- if ne $iEndpoint (sub $replicaCount 1) }}
+                {{- $endpointStr = printf "%s," $endpointStr }}
+                {{- end }}
+                {{- end }}
+                {{- $endpointStr = printf "%s]" $endpointStr }}
+                - {{ $endpointStr }}
```

Comment:
@mikyll @juan131 

This only work if the cluster is completly unauthenticated. Nowhere here is set the user or password options